nameko_grpc/entrypoint.py

```diff
@@ -261,18 +264,28 @@ def handle_request(self, request_stream, response_stream):
     def handle_result(self, response_stream, worker_ctx, result, exc_info):
 
         if self.cardinality in (Cardinality.STREAM_UNARY, Cardinality.UNARY_UNARY):
-            result = (result,)
+            response = (result,)
+        else:
+            result = Teeable(result)
+            response = result
 
         if exc_info is None:
-            try:
-                response_stream.populate(result)
-            except Exception as exception:
-                error = GrpcError(
-                    status=StatusCode.UNKNOWN,
-                    details="Exception iterating responses: {}".format(exception),
-                    debug_error_string="<traceback>",
-                )
-                response_stream.close(error)
+
+            def send_response():
+                try:
+                    response_stream.populate(response)
+                except Exception as exception:
+                    error = GrpcError(
+                        status=StatusCode.UNKNOWN,
+                        details="Exception iterating responses: {}".format(exception),
+                        debug_error_string="<traceback>",
+                    )
+                    response_stream.close(error)
+
+            self.container.spawn_managed_thread(
+                send_response, identifier="send_response"
+            )
```

Comment:
Send response had to be deferred to a thread, otherwise the `response` iterator was exhausted before `worker_result` was called in any `DependencyProviders`